pkg/controller/garbagecollector/operations.go

```diff
@@ -49,13 +49,16 @@ func (gc *GarbageCollector) apiResource(apiVersion, kind string) (schema.GroupVe
 	return mapping.Resource, mapping.Scope == meta.RESTScopeNamespace, nil
 }
 
-func (gc *GarbageCollector) deleteObject(item objectReference, policy *metav1.DeletionPropagation) error {
+func (gc *GarbageCollector) deleteObject(item objectReference, resourceVersion string, policy *metav1.DeletionPropagation) error {
 	resource, namespaced, err := gc.apiResource(item.APIVersion, item.Kind)
 	if err != nil {
 		return err
 	}
 	uid := item.UID
 	preconditions := metav1.Preconditions{UID: &uid}
+	if len(resourceVersion) > 0 {
+		preconditions.ResourceVersion = &resourceVersion
```

Comment:
`attemptToDeleteItem()` errors are influencing the controlling behavior inside the `attemptToDeleteWorker`

naive question, what does this change means in terms of the returning errors?

if it tries to delete some object that is more recent is an error?

Reply:
yeah, it returns a conflict error

Reply:
then we should handle it here too, no?

https://github.com/kubernetes/kubernetes/blob/c549b59983e114a872b0df18d74c1d217f3f62bd/pkg/controller/garbagecollector/garbagecollector.go#L361-L389

Reply:
we'll fall through this bit and retry, which seems correct:

https://github.com/kubernetes/kubernetes/blob/c549b59983e114a872b0df18d74c1d217f3f62bd/pkg/controller/garbagecollector/garbagecollector.go#L378-L382

we could specifically handle a conflict and lower the verbosity of what we log